### ISSUE TITLE
Autocompletion stops on perfect match

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#56270]
### Godot version

28174d5

### System information

W10

### Issue description

When you have a word with perfect match to some identifier, the autocompletion will no longer appear.

Godot
![image](https://user-images.githubusercontent.com/2223172/147423057-0372501e-3caa-48b4-a4d6-11470b683e8c.png)

VS Code (expected)
![image](https://user-images.githubusercontent.com/2223172/147423075-b54bd700-be77-4729-a9ee-a8bb6ddfd654.png)

At least using Ctrl + Space should force the popup.

### Steps to reproduce

1. Type something that perfectly matches some identifier, but also is part of another, longer identifier
2. Try to autocomplete the longer identifier

**Comment by PhoenixFlame101** [comment]
Not able to reproduce after building from the provided commit (`Godot Engine v4.0.dev.custom_build.28174d531`)

Screenshot:
![image](https://user-images.githubusercontent.com/51199011/147538401-4e96e6e1-189f-4779-8464-c0e2ab3ddfa9.png)

Another Screenshot:
![image](https://user-images.githubusercontent.com/51199011/147538601-d97bf48e-f37a-4409-8745-29839b414f7e.png)

Ctrl+Space also reopens the autocomplete if you click out.

Seems to be that declared variables do not show up at all. Maybe you should open up another issue regarding this.
![image](https://user-images.githubusercontent.com/51199011/147539008-9e7717b0-6610-4eff-8ee7-e67db05a40c8.png)

**Comment by KoBeWi** [comment]
It stops on `var`, but shows correctly on `vari`
![image](https://user-images.githubusercontent.com/2223172/147558328-b1caeff1-f836-43c8-9b83-8967d07f64f9.png)
I've seen this happen in some other cases too. Maybe it happens only to keywords?

**Comment by PhoenixFlame101** [comment]
I am unable to get autocomplete to work when typing at the top of the file, even after pressing Ctrl+Space, only after typing something previously in the same line like `var varname =` (as I did in the previous comment).

Trying this inside a function (`_ready`), as you seem to have done, I am able to reproduce the error as you describe it.

But the  peculiar thing is that if the autocomplete is already open while you are typing (if you type slowly and it opens up at `va` for example), then the autocomplete works as intended.

![image](https://user-images.githubusercontent.com/51199011/147578752-a159590d-e6bc-493e-bb12-6c589fefaf28.png)

I am able to reproduce the error for other keywords like `const` and `return`, but the intended behavior is observed for `func`, `signal`, `static` etc. I guess someone with more knowledge on the nuances of the language can chime in and explain why this is the case (and how the issue can be fixed ;D).
